src/utils/timestampUtils.tsx

```diff
@@ -0,0 +1,94 @@
+import TimeAgo from "javascript-time-ago";
+import en from "javascript-time-ago/locale/en";
+import { ReduxMessage } from "./../store/reducers/messageListReducer";
+import * as _ from "lodash";
+
+const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+TimeAgo.addLocale(en);
+const timeAgo = new TimeAgo(userTimezone);
+
+function formatMessageTime(
+  dateString: Date,
+  lastMessage: boolean,
+  on: boolean
+) {
+  const date = new Date(dateString);
+  const currentDate = new Date();
+
+  if (lastMessage && currentDate.toDateString() !== date.toDateString()) {
+    // If it's not the same day, show date.
+    return date.toDateString();
+  } else if (currentDate.getTime() - date.getTime() >= 3 * 60 * 60 * 1000) {
+    // If it's older than 3 hours, return time based on the 'on' const
+    const hours = date.getHours();
+    const minutes = date.getMinutes();
+
+    if (on) {
+      // 24-hour format
+      const formattedHours = hours.toString().padStart(2, "0");
+      const formattedMinutes = minutes.toString().padStart(2, "0");
```

Comment:
There is definitely a standard time formatting function that can do it, why do we have to perform these standard operations manually?

Ideally, a `toLocaleDateString()` and `toLocaleTimeString()` with proper locale should suffice even without using a 24h-time toggle.